### ISSUE TITLE
Problems with p6doc build 

### ISSUE THREAD

**Issue by JJ** [perl6/doc#2387]
## The problem

Reported in [the perl6 mailing list](https://www.nntp.perl.org/group/perl.perl6.users/2018/10/msg6223.html):

```
Attempting to build the P6doc index produced the following result:
-
$ p6doc build
Too many positionals passed; expected 1 argument but got 2
  in sub build_index at
/home/guru/rakudo/rakudo-star-2018.06/install/share/perl6/site/resources/C8DD97308D226DDE4AA60EA017CB6BB59F9AA625
line 193
  in sub MAIN at
/home/guru/rakudo/rakudo-star-2018.06/install/share/perl6/site/resources/C8DD97308D226DDE4AA60EA017CB6BB59F9AA625
line 163
  in block <unit> at
/home/guru/rakudo/rakudo-star-2018.06/install/share/perl6/site/resources/C8DD97308D226DDE4AA60EA017CB6BB59F9AA625
line 316
  in sub MAIN at
/home/guru/rakudo/rakudo-star-2018.06/install/share/perl6/site/bin/p6doc
line 2
  in block <unit> at
/home/guru/rakudo/rakudo-star-2018.06/install/share/perl6/site/bin/p6doc
line 2

$ uname -a
Linux localhost 4.14.70-desktop-2.mga6 #1 SMP Thu Sep 20 22:05:46 UTC
2018 x86_64 x86_64 x86_64 GNU/Linux

[]$ perl6 -v
This is Rakudo Star version 2018.06 built on MoarVM version 2018.06
implementing Perl 6.c.

(BTW, it appears that perl-all no longer works. I sent this message
there, and saw nothing.)
```

## Suggestions

Check the error and report back to the mailing list.

**Comment by stmuk** [comment]
Can't reproduce

**Comment by coke** [comment]
Also unable to duplicate. closing ticket.

**Comment by agurB** [comment]
**same here** 
 p6doc -f build
building index on first run. Please wait...
Too many positionals passed; expected 1 argument but got 2
  in sub build_index at /usr/lib64/perl6/site/resources/C8DD97308D226DDE4AA60EA017CB6BB59F9AA625 line 193
  in sub MAIN at /usr/lib64/perl6/site/resources/C8DD97308D226DDE4AA60EA017CB6BB59F9AA625 line 136
  in block <unit> at /usr/lib64/perl6/site/resources/C8DD97308D226DDE4AA60EA017CB6BB59F9AA625 line 316
  in sub MAIN at /usr/lib64/perl6/site/bin/p6doc line 2
  in block <unit> at /usr/lib64/perl6/site/bin/p6doc line 2

perl6 -version
This is Rakudo version 2018.04 built on MoarVM version 2018.04.1
implementing Perl 6.c.

Fedora28 4.19.2-200.fc28.x86_64

**Comment by stmuk** [comment]
@agurB Are you using a Fedora RPM or have you installed from source?

**Comment by agurB** [comment]
dnf install rakudo
dnf install rakudo-zef
zef install p6doc

**Comment by agurB** [comment]
installed from source works fine
Fedora 4.19.5-300.fc29.x86_64 #1 SMP Tue Nov 27 19:29:23 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by coke** [comment]
Given that this was fixed in HEAD at some point, I'm closing the ticket.

**Comment by alanrocker** [comment]
The problem has recurred with:
perl6 -v
This is Rakudo Star version 2019.03.1 built on MoarVM version 2019.03
implementing Perl 6.d.
uname -a
Linux localhost 5.2.7-desktop-1.mga7 #1 SMP Wed Aug 7 10:32:19 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Comment by alanrocker** [comment]
It looks as though whatever was "fixed in HEAD" hasn't made it into Rakudo*. Could we have details (or at least an issue number) to help implement the fix, please?

**Comment by finanalyst** [comment]
We have a new p6doc after GSoC. Please look at https://github.com/noisegul/perl6-p6doc

This work uses another GSoC project to remake Perl6-Documentable.  
The two are part of a major refactor of the documentation system that splits out various parts of the perl6-doc repo as described in the wiki.   
The refactoring has improved tests and a more logical robust structure, whilst at the same time ensuring that the user facing pages are the same.
In short, the problems facing p6doc have been resolved, but the refactoring requires systematic and stage introduction. The plan is to do this in September. A little patience please.

**Comment by alanrocker** [comment]
If there's anything I can do to help with the Rakudo* update, I'd be happy to do it.

**Comment by alanrocker** [comment]
What was the solution that supposedly made it into HEAD?

**Comment by JJ** [comment]
@alanrocker p6doc is an almost complete rewrite. So I don't think this is addressed per se.

**Comment by coke** [comment]
With p6doc spun off, what's left of this issue in this repository?
